vscode/vscode/vscode.ml

```diff
@@ -1898,6 +1920,48 @@ module DocumentFormattingEditProvider = struct
       [@@js.builder]]
 end
 
+module Hover = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val contents : t -> MarkdownString.t [@@js.get]
+
+    val range : t -> Range.t [@@js.get]
+
+    val make :
+         contents:
+           ([ `MarkdownString of MarkdownString.t
+            | `MarkdownStringArray of MarkdownString.t list
+            ]
+           [@js.union])
+      -> t
+      [@@js.new "vscode.Hover"]]
+end
+
+module HoverProvider = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val provideHover :
+         t
+      -> document:TextDocument.t
+      -> position:Position.t
+      -> token:CancellationToken.t
+      -> Hover.t list ProviderResult.t
```

Comment:
Isn't this is supposed to return a single `Hover` value? There is no `[]` or `Array< >` in the TypeScript code

https://github.com/microsoft/vscode/blob/1.58.2/src/vs/vscode.d.ts#L2674

Reply:
Done in 84ec3948dc80307b424fa225aebf385e1fe7e03e